src/utilities/tests/use-is-mounted-ref.test.tsx

```diff
@@ -0,0 +1,56 @@
+import React, {useEffect} from 'react';
+import {mount} from 'test-utilities';
+import {useIsMountedRef} from '../use-is-mounted-ref';
+
+describe('useIsMountedRef', () => {
+  it('returns a false value before mounting', () => {
+    const spy = jest.fn((_) => {});
+
+    function MockComponent() {
+      const isMounted = useIsMountedRef();
+      spy(isMounted.current);
+      return null;
+    }
+
+    mount(<MockComponent />);
+    expect(spy).toHaveBeenCalledWith(false);
+  });
+
+  it('returns a false value after unmounting', () => {
+    const spy = jest.fn((_) => {});
+
+    function MockComponent() {
+      const isMounted = useIsMountedRef();
+
+      useEffect(() => {
+        return () => {
+          // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
dom nodes need to have their references saved since react could have gotten rid of them before the cleanup happens but this is 👌

Reply:
can we add a comment inline that says what's being excluded and why?